py-rattler/src/networking/middleware.rs

```diff
@@ -0,0 +1,70 @@
+use pyo3::{pyclass, pymethods, FromPyObject};
+use rattler_networking::{
+    mirror_middleware::Mirror, AuthenticationMiddleware, AuthenticationStorage, MirrorMiddleware,
+};
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(FromPyObject)]
+pub enum PyMiddleware {
+    MirrorMiddleware(PyMirrorMiddleware),
+    AuthenticationMiddleware(PyAuthenticationMiddleware),
+}
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyMirrorMiddleware {
+    pub(crate) inner: HashMap<Url, Vec<Mirror>>,
+}
+
+#[pymethods]
+impl PyMirrorMiddleware {
+    #[new]
+    pub fn __init__(inner: HashMap<String, Vec<String>>) -> Self {
+        let map = inner
+            .into_iter()
+            .map(|(k, v)| {
+                (
+                    Url::parse(&k).unwrap(),
+                    v.into_iter()
+                        .map(|url| Mirror {
+                            url: Url::parse(&url).unwrap(),
+                            no_zstd: false,
+                            no_bz2: false,
+                            no_jlap: false,
+                            max_failures: None,
```

Comment:
Are these options irrelevant for defining mirrors? Would it maybe make sense to create another struct to pass them from Python?

Reply:
i would say they are irrelevant until somebody actually needs them in python.
then, the API can be extended IMO but in order to keep py-rattler somewhat simple i would suggest we leave them like this for now